lib/confex_consul.ex

```diff
@@ -3,30 +3,33 @@ defmodule ConfexConsul do
   Documentation for `ConfexConsul`.
   """
 
+  require Logger
+
   @behaviour Confex.Adapter
 
   @impl Confex.Adapter
-  def fetch_value(consul_key) do
-    case ConfexConsul.LocalCache.get(consul_key) do
+  def fetch_value(key) do
+    case ConfexConsul.LocalCache.get(key) do
       {:hit, value} -> value
-      :miss -> get_from_consul_and_cache(consul_key)
+      :miss -> get_from_consul_and_cache(key)
     end
   end
 
   @doc false
-  defp get_from_consul_and_cache(consul_key) do
-    consul_key
-    |> ConfexConsul.ConsulKv.get_value()
-    |> write_cache_and_return(consul_key)
-  end
+  defp get_from_consul_and_cache(key) do
+    with {:ok, value} <- ConfexConsul.ConsulKv.get_value(key) do
+      write_cache_and_return(key, {:ok, value})
```

Comment:
It's not a good idea to use the result of one write operation function.

How about:

```suggestion
    with {:ok, value} <- ConfexConsul.ConsulKv.get_value(key),
         _ <- write_cache(key, value) do
      {:ok, value}
```